RHDEVDOCS-3918 - Document Alertmanager for user workload monitoring (#46357)
openshift/openshift-docs: modules/monitoring-enabling-a-separate-alertmanager-instance-for-user-defined-projects.adoc

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` role.
* You have enabled monitoring for user-defined projects in the `cluster-monitoring-config` config map for the `openshift-monitoring` namespace.
* You have installed the OpenShift CLI (`oc`).

.Procedure

. Edit the `user-workload-monitoring-config` `ConfigMap` object:
+
[source,terminal]
----
$ oc -n openshift-user-workload-monitoring edit configmap user-workload-monitoring-config
----
+
. Add `enabled: true` and `enableAlertmanagerConfig: true` in the `alertmanager` section under `data/config.yaml`:
+
[source,yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: user-workload-monitoring-config
  namespace: openshift-user-workload-monitoring
data:
  config.yaml: |
    alertmanager: 
      enabled: true <1>
      enableAlertmanagerConfig: true <2>
----
<1> Set the `enabled` value to `true` to enable a dedicated instance of the Alertmanager for user-defined projects in a cluster. Set the value to `false` or omit the key entirely to disable the Alertmanager for user-defined projects. 

> If you set this value to `false` or if the key is omitted, user-defined alerts are routed to the default platform Alertmanager instance

Reviewer: missing period?

Author: fixed

> <2> Set the `enableAlertmanagerConfig` value to `true` to enable users to define their own alert routing configurations that use the dedicated instance of Alertmanager rather than the platform instance.

Reviewer: ```suggestion
<2> Set the `enableAlertmanagerConfig` value to `true` to enable users to define their own alert routing configurations with `AlertmanagerConfig` objects.
```

Author: Changed per your suggestion

// Module included in the following assemblies:
//
// * monitoring/enabling-alert-routing-for-user-defined-projects.adoc

:_content-type: PROCEDURE
[id="monitoring-enabling-a-separate-alertmanager-instance-for-user-defined-projects_{context}"]
= Enabling a separate Alertmanager instance for user-defined projects
 
In some clusters, you might not want to use the default platform instance of Alertmanager for user-defined alerts. 

> In these cases, you can optionally enable a dedicated instance of Alertmanager to send alert notifications for user-defined projects in a cluster.

Reviewer 2: The juxtaposition of "user-defined alerts" in line 9 and "alert notifications for user-defined projects in a cluster" in line 10 is raises questions. Are the two equivalent or distinct?

Author: Edited to read `In some clusters, you might not want to use the default platform instance of Alertmanager to send alert notifications for user-defined projects. 
In these cases, you can optionally enable a dedicated instance of Alertmanager to send alerts for these projects.`

Enabling a separate Alertmanager instance dedicated to user-defined projects offers these benefits:

* Developers and other users of user-defined projects can create user-defined alerts and alert routing without the help of an administrator.
* A dedicated Alertmanager instance for user-defined projects can help reduce the load on the default platform Alertmanager instance.

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` role.
* You have enabled monitoring for user-defined projects in the `cluster-monitoring-config` config map for the `openshift-monitoring` namespace.
* You have installed the OpenShift CLI (`oc`).

.Procedure

. Edit the `user-workload-monitoring-config` `ConfigMap` object:
+
[source,terminal]
----
$ oc -n openshift-user-workload-monitoring edit configmap user-workload-monitoring-config
----
+

> . Add `enabled: true` and `enableAlertmanagerConfig: true` under `alertmanager` under `data/config.yaml`:

Reviewer 3: Add `enabled: true`  and `enableAlertmanagerConfig: true` for the `alertmanager` component under `data/config.yaml`:

Author: Changed per your suggestion.

+
[source,yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: user-workload-monitoring-config
  namespace: openshift-user-workload-monitoring
data:
  config.yaml: |
    alertmanager: 
      enabled: true <1>
      enableAlertmanagerConfig: true <2>
----

> <1> Set the `enabled` value to `true` to enable a dedicated instance of the Alertmanager for user-defined projects in a cluster. Set the value to `false` or omit the key entirely to disable the Alertmanager for user-defined projects.

Author: Changed to:

`If you set this value to `false` or if the key is omitted, user-defined alerts are routed to the default platform Alertmanager instance`

// Module included in the following assemblies:
//
// * monitoring/enabling-alert-routing-for-user-defined-projects.adoc

:_content-type: PROCEDURE
[id="monitoring-enabling-a-separate-alertmanager-instance-for-user-defined-projects_{context}"]
= Enabling a separate Alertmanager instance for user-defined projects
 
In some clusters, you might not want to use the default platform instance of Alertmanager for user-defined alerts. 
In these cases, you can optionally enable a dedicated instance of Alertmanager to send alert notifications for user-defined projects in a cluster.

Enabling a separate Alertmanager instance dedicated to user-defined projects offers these benefits:

* Developers and other users of user-defined projects can create user-defined alerts and alert routing without the help of an administrator.
* A dedicated Alertmanager instance for user-defined projects can help reduce the load on the default platform Alertmanager instance.

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` role.

> * You have enabled monitoring for user-defined projects in the `cluster-monitoring-config` config map in the `openshift-monitoring` namespace.

Reviewer 3: `in the `cluster-monitoring-config` config map in the `openshift-monitoring` namespace`
`in the ** config map in the ** namespace`, the expression seems not good

* You have installed the OpenShift CLI (`oc`).

.Procedure

. Edit the `user-workload-monitoring-config` `ConfigMap` object:
+
[source,terminal]
----
$ oc -n openshift-user-workload-monitoring edit configmap user-workload-monitoring-config
----
+
. Add `enabled: true` and `enableAlertmanagerConfig: true` under `alertmanager` under `data/config.yaml`:
+
[source,yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: user-workload-monitoring-config
  namespace: openshift-user-workload-monitoring
data:
  config.yaml: |

>     enableUserWorkload: true

Reviewer 3: `enableUserWorkload: true` should be removed, since it is enabled  in the cluster-monitoring-config config map under the openshift-monitoring namespace, the Prerequisites already enabled user projects monitoring
`You have enabled monitoring for user-defined projects in the cluster-monitoring-config config map in the openshift-monitoring namespace.`

Author: removed

    alertmanager: 
      enabled: true <1>
      enableAlertmanagerConfig: true <2>
----
<1> Set the `enabled` value to `true` to enable a dedicated instance of the Alertmanager for user-defined projects in a cluster. Set the value to `false` or omit the key entirely to disable the Alertmanager for user-defined projects.

> <2> Set the `enableAlertmanagerConfig` value to `true` to allow users to create user-defined alerts.

Reviewer 3: Set the `enableAlertmanagerConfig` value to `true` to allow users to create user-defined alert routing configurations that use the dedicated platform instance of Alertmanager.
